core/pkg/opencost/allocation.go

```diff
@@ -198,8 +255,16 @@ func (r *RawAllocationOnlyData) Equal(that *RawAllocationOnlyData) bool {
 	if r == nil || that == nil {
 		return false
 	}
-	return util.IsApproximately(r.CPUCoreUsageMax, that.CPUCoreUsageMax) &&
+	cmpResult := util.IsApproximately(r.CPUCoreUsageMax, that.CPUCoreUsageMax) &&
 		util.IsApproximately(r.RAMBytesUsageMax, that.RAMBytesUsageMax)
+
+	if r.GPUUsageMax != nil && that.GPUUsageMax != nil {
+		cmpResult = cmpResult && util.IsApproximately(*r.GPUUsageMax, *that.GPUUsageMax)
+	} else {
+		cmpResult = false
+	}
```

Comment:
So, I think you would still want equality to be true if both GPUUsageMax fields are nil? ie: If both fields are nil, we can just leave `cmpResult` alone.

```
if r.GPUUsageMax != nil && that.GPUUsageMax != nil {
    cmpResult = cmpResult && util.IsApproximately(*r.GPUUsageMax, *that.GPUUsageMax)
} else if !(r.GPUUsageMax == nil && that.GPUUsageMax == nil) {
    cmpResult = false
}
```

Reply:
Great unit test case :)